app/concepts/tickets/representers/single.rb

```diff
@@ -0,0 +1,27 @@
+module Tickets
+  module Representers
+    class Single
+      attr_reader :ticket
+
+      def initialize(ticket)
+        @ticket = ticket
+      end
+
+      def basic
+        {
+          id: ticket.id,
+          seat: ticket.seat,
+          price: ticket.price,
+          ticket_type: ticket.ticket_type,
+          hall: cinema_hall.name
+        }
+      end
+
+      private
+
+      def cinema_hall
+        CinemaHalls::Repository.new.find(ticket.cinema_hall_id)
```

Comment:
Could you please help me to improve this code, @w0st? I want to display hall's name inside a ticket but I display tickets by two routes  (single ticket) `/ticket_desks/1/reservations/1/ticket/1` and (all reservation's tickets) `/ticket_desks/1/reservations/1`. Displaying several tickets calls `CinemaHalls::Repository` many times. Is it acceptable? or I should initialize `Tickets::Representer` with cinema_hall and call `CinemaHalls::Repository` inside each controller (reservations and tickets) instead of `Tickets::Representer`?

Reply:
I think doing something like that:
```suggestion
        ticket.cinema_hall
```
would be enough here - you don't really need to `find` it - it belongs to an hall, so this relation exist here :) 
And about N+1 problem - when you are using `Tickets::Representers::All` you can to the existing relation add [`includes(...)`](https://apidock.com/rails/v6.1.3.1/ActiveRecord/QueryMethods/includes) - that way it will preload `cinema_hall` for each ticket, so it will not generate any extra queries.